src/realm/transaction.hpp

```diff
@@ -484,7 +485,10 @@ inline bool Transaction::internal_advance_read(O* observer, VersionID version_id
 
     // Synchronize readers view of the file
     SlabAlloc& alloc = m_alloc;
-    alloc.update_reader_view(new_file_size);
+    {
+        util::CheckedLockGuard lock_guard(db->m_mutex);
```

Comment:
There appears to be an assortment of things which are guarded by DB::m_mutex now, but it's not really clear to me what is or why? The SlabAlloc functions don't document that they need to be called with a lock held.

DB::m_mutex being an InterprocessMutex means that we have to be extremely careful with where we acquire it. On IOS it's *extremely* slow, and locking it will kill the process if the app is currently running in the background. Together this means that we can't safely lock it on reads.

Reply:
right good catch, locking db->mutex is no longer needed with this new approach